low-level/files/opt/seagate/sspl/setup/manifest_support_bundle

```diff
@@ -144,15 +144,104 @@ class ManifestSupportBundle(object):
 
         print("Manifest Support Bundle Location: %s" % self.localTempPath+file_name)
 
+    def capture_node_data(self):
+        # Get Node Support Data
+        node_support_data = {}
+        node_support_data.update(self.support_common_data)
+        node_support_data.update({"hostname": self.hostname})
+        node_support_data["hw_specifics"] = self.build_node_hw_specifics_instances()
+        self.write_content_in_file(node_support_data, self.node_manifest,
+                                   'node support data')
+        print('Node Support data Json Created Successfully..! :%s' % self.node_manifest)
+
+    def capture_enclosure_data(self):
+        # Get Enclosure Support Data
+        encl_support_data = {}
+        encl_support_data.update(self.support_common_data)
+        encl_data_response = self.build_encl_hw_specifics_instances()
+        if encl_data_response and 'enclosures' in encl_data_response:
+            encl_support_data["enclosure_wwn"] = encl_data_response["enclosures"][0]["enclosure-wwn"]
+            encl_support_data["hw_specifics"] = encl_data_response
+            self.write_content_in_file(encl_support_data, self.encl_manifest,
+                                        'enclosure support data')
+            print('Enclosure Support data Json Created Successfully..! :%s' % self.encl_manifest)
+        else:
+            print("Management controller IPs are not configured properly on node or in ConfStore", end=" ")
+            print("or it is inaccessible for the current node.")
+            print("Ignored to collect enclosure manifest data.")
+
+    def build_node_hw_specifics_instances(self):
+        lshw_dict = {}
+        with tempfile.TemporaryFile() as tempf:
+            proc = subprocess.Popen(['lshw', '-json'], stdout=tempf)
```

Comment:
Instead of creating a temp file which is a overhead, you can directly capture the output.
You can use `out, err = proc.communicate()`

Reply:
Done